R/utils.R

```diff
@@ -127,3 +127,24 @@ flatten_list <- function(x) {
         return(x)
     }
 }
+
+# This function gets called on package load.
+.onLoad <- function(libname, pkgname) {
+    # If credentials are in environment variables, use those.
+    creds <- aws.signature::locate_credentials()
+    if (!all(is.null(creds$key), is.null(creds$secret))) {
+        return
+    }
+    # Load default AWS credentials so calls to S3 'just work', allowing
+    # the package to behave like the AWS CLI
+    if (nchar(Sys.getenv("AWS_PROFILE"))) {
+        aws.signature::use_credentials(Sys.getenv("AWS_PROFILE"))
+    } else {
+        aws.signature::use_credentials()
+    }
+}
+
+# Handle when package is attached (invoked with pkg::function()) not loaded
```

Comment:
You don't need this function. `.onLoad()` is called when a package namespace is loaded, which occurs either at `library("aws.s3")` or `aws.s3::{anything}()`. `.onAttach()` is called only when `library()` or `require()` is used and is discouraged: https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#index-_002eonAttach